src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -135,20 +128,27 @@ private void OnBindingStateChanged(object sender, BindingStateEventArgs e)
 
         private void RaiseAnalyzersChangedIfBindingChanged(bool? isBindingCleared = null)
         {
-            var newBindingConfiguration = this.configurationProvider.GetConfiguration();
+            var newBindingConfiguration = configurationProvider.GetConfiguration();
 
             if (!CurrentConfiguration.Equals(newBindingConfiguration))
             {
                 CurrentConfiguration = newBindingConfiguration;
-                this.SolutionBindingChanged?.Invoke(this, new ActiveSolutionBindingEventArgs(newBindingConfiguration));
+                SolutionBindingChanged?.Invoke(this, new ActiveSolutionBindingEventArgs(newBindingConfiguration));
             }
             else if (isBindingCleared == false)
             {
                 SolutionBindingUpdated?.Invoke(this, EventArgs.Empty);
             }
-            else
+
+            SetBoundSolutionUiContext();
+        }
+
+        private void SetBoundSolutionUiContext()
+        {
+            if (vsMonitorSelection.GetCmdUIContextCookie(ref BoundSolutionVsUiContext.Guid, out var cookie) == VSConstants.S_OK)
```

Comment:
The VS [documentation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.interop.ivsmonitorselection.getcmduicontextcookie?view=visualstudiosdk-2019) suggests that `GetCmdUIContextCookie` should only be called once per GUID, implying the caller should hold on to the cookie and re-use it.
The behaviour of calling the method multiple times doesn't seem to be documented.